### PR TITLE
ci: Make integrations tests more stable

### LINKED ISSUE #301: Make integrations tests more stable
Our integration tests are highly unstable, often requiring multiple attempts to pass. Below are the two main issues we are encountering...

### Issues during Actor creation

The first problem occurs when creating an Actor in `conftest/make_actor` ([conftest.py#L247](https://github.com/apify/apify-sdk-python/blob/v2.0.1b9/tests/integration/conftest.py#L247)):

```python
    ...
    build = await actor_client.build(version_number='0.0', wait_for_finish=300)
    assert build['status'] == ActorJobStatus.SUCCEEDED
```

The test frequently fails with the following assertion error:

```
>       assert build['status'] == ActorJobStatus.SUCCEEDED
E       AssertionError: assert 'READY' == <ActorJobStat...: 'SUCCEEDED'>
E         
E         - SUCCEEDED
E         + READY
```

### Issues during Actor execution

Another common issue arises when checking the Actor's run status in the integration tests:

```python
    ...
    run_result = await actor.call()
    assert run_result is not None
    assert run_result['status'] == 'SUCCEEDED'
```

Here, the test often fails with a similar mismatch in the expected status:

```
>       assert build['status'] == ActorJobStatus.SUCCEEDED
E       AssertionError: assert 'READY' == <ActorJobStat...: 'SUCCEEDED'>
E         
E         - SUCCEEDED
E         + READY
```

### PR DESCRIPTION
**Pull request by vdusek**
Both the build Actor step and the run Actor step.

Closes #301

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -56,6 +56,53 @@ def apify_client_async() -> ApifyClientAsync:
...
 
 
+class RunActorFunction(Protocol):
+    """A type for the `run_actor` fixture."""
+
+    def __call__(
+        self,
+        actor: ActorClientAsync,
+        *,
+        run_input: Any = None,
```
Comment (janbuchar) [inline tests/integration/conftest.py:66]: Isn't `run_input` a dict?

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -56,6 +56,53 @@ def apify_client_async() -> ApifyClientAsync:
...
+
+        assert isinstance(call_result, dict), 'The result of ActorClientAsync.call() is not a dictionary.'
+        assert 'id' in call_result, 'The result of ActorClientAsync.call() does not contain an ID.'
+
+        run_client = apify_client_async.run(call_result['id'])
+        run_result = await run_client.wait_for_finish(wait_secs=600)
+
+        assert isinstance(run_result, dict), 'The result of RunClientAsync.wait_for_finish() is not a dictionary.'
+        assert 'status' in run_result, 'The result of RunClientAsync.wait_for_finish() does not contain a status.'
+        return run_result
```
Comment (janbuchar) [inline tests/integration/conftest.py:101]: It is quite an intrusive change, but couldn't we do `from apify._models import ActorRun` and `return ActorRun.model_validate(run_result)` here?

**Review by janbuchar — COMMENTED** [review]
I like this, just one idea. Feel free to dismiss it if you like :slightly_smiling_face:

File: tests/integration/conftest.py (hunk 2 above)
Comment (vdusek) [reply tests/integration/conftest.py:101]: Hmm, let's try it

File: tests/integration/conftest.py (hunk 1 above)
Comment (vdusek) [reply tests/integration/conftest.py]: It really is `Any` - https://github.com/apify/apify-client-python/blob/v1.8.1/src/apify_client/clients/resource_clients/actor.py#L591

Comment (janbuchar) [reply tests/integration/conftest.py]: Huh, weird, but not important.

**Review by janbuchar — APPROVED** [review]
LGTM

**Pull request merged**